samcli/commands/pipeline/init/interactive_init_flow.py

```diff
@@ -192,7 +198,7 @@ def _generate_from_pipeline_template(self, pipeline_template_dir: Path) -> List[
         and return the list of generated files.
         """
         pipeline_template: Template = _initialize_pipeline_template(pipeline_template_dir)
-        required_env_number = 2  # TODO: read from template
+        required_env_number = (pipeline_template.metadata or {}).get("number_of_stages", 2)
```

Comment:
Can we add a debug log here mentioning that cannot parse metadata.json so defaulting to 2?

Reply:
added